File: src/p0009/rust/solution.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        if x > 0 && x % 10 == 0 {
            return false;
        }
        let (mut x, mut y) = (x, 0);
        while y < x {
            y = y * 10 + x % 10;
            if x == y { break; }
            x /= 10;
        }
        return x == y;
    }
}

File: src/p0009/golang/solution.go
package main

func isPalindrome(x int) bool {
	if x > 0 && x % 10 == 0 {
		return false
	}
	y := 0
	for ; y < x; x /= 10 {
		y = y * 10 + x % 10
		if x == y {
			break
		}
	}
	return x == y
}
